providers/nextcloud.go

```diff
@@ -23,17 +24,49 @@ func NewNextcloudProvider(p *ProviderData) *NextcloudProvider {
 	return &NextcloudProvider{ProviderData: p}
 }
 
-// GetEmailAddress returns the Account email address
-func (p *NextcloudProvider) GetEmailAddress(ctx context.Context, s *sessions.SessionState) (string, error) {
+// EnrichSession is called after Redeem to allow providers to enrich session fields
+// such as User, Email, Groups with provider specific API calls.
+func (p *NextcloudProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
 	json, err := requests.New(p.ValidateURL.String()).
 		WithContext(ctx).
 		WithHeaders(makeOIDCHeader(s.AccessToken)).
 		Do().
 		UnmarshalJSON()
 	if err != nil {
-		return "", fmt.Errorf("error making request: %v", err)
+		return fmt.Errorf("error requesting ocs data: %v", err)
 	}
 
-	email, err := json.Get("ocs").Get("data").Get("email").String()
-	return email, err
+	data := json.Get("ocs").Get("data")
```

Comment:
NIT: helper method is called `populateSessionFromOcs` -- should you pass the JSON to that function & do the extraction inside it?